pkg/terraform/terraform.go

```diff
@@ -30,7 +31,7 @@ func initActionAutoApprove(terraformClientPath string, tfAction, tfEntrypoint st
 		return err
 	}
 
-	err = pkg.ExecShellWithVars(tfEnvs, terraformClientPath, tfAction, "-auto-approve")
+	err = pkg.ExecShellWithVars(tfEnvs, terraformClientPath, tfAction, "-auto-approve", fmt.Sprintf("-parallelism=%d", runtime.NumCPU()*2))
```

Comment:
Since all modern computers these days support SMT then we know that the cores available to the operating system multiplied by 2 is the maximum number of threads we can process. So the computation at the very end should in fact prevent us from over requesting. Additionally, if we did overrequest, terraform doesn't break execution it just runs a little slower because it divided the work into too many threads and has to wait for the work to get juggled by the available cpus. 

I've never seen it break execution.